tools/code-owners-parser/CodeOwnersParser/GlobFilePath.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.FileSystemGlobbing;
+
+namespace Azure.Sdk.Tools.CodeOwnersParser;
+
+public class GlobFilePath
+{
+    private readonly string filePath;
+
+    public GlobFilePath(string globFilePath)
+    {
+        Debug.Assert(globFilePath.IsGlobFilePath());
+        this.filePath = globFilePath;
+    }
+
+    public static bool IsGlobFilePath(string path)
+        => path.Contains('*');
```

Comment:
Is that the only special char?

Reply:
Yes, per these docs:
- https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.filesystemglobbing.matcher?view=dotnet-plat-ext-7.0#remarks
- https://learn.microsoft.com/en-us/dotnet/core/extensions/file-globbing#pattern-formats

I will add these links to the comments to eliminate the guesswork.